src-gui/src/main/java/org/openkilda/security/CustomRestTemplateConfiguration.java

```diff
@@ -0,0 +1,52 @@
+package org.openkilda.security;
+
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.Resource;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+
+import javax.net.ssl.SSLContext;
+
+
+@Configuration
+public class CustomRestTemplateConfiguration {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CustomRestTemplateConfiguration.class);
+
+    @Value("${victoria.trust.store}")
+    private Resource trustStore;
+
+    @Value("${victoria.trust.store.password}")
+    private String trustStorePassword;
+
+    @Bean
+    public RestTemplate restTemplate() {
+        try {
+            LOGGER.info("Configuring restTemplate with the trustStore:{}, and passord:{}", trustStore.getURL(), trustStorePassword);
+            SSLContext sslContext = new SSLContextBuilder()
+                    .loadTrustMaterial(trustStore.getURL(), trustStorePassword.toCharArray()).build();
+            LOGGER.info("sslContext has been initialized");
+            SSLConnectionSocketFactory sslConFactory = new SSLConnectionSocketFactory(sslContext);
+            LOGGER.info("sslConFactory has been initialized");
+            CloseableHttpClient httpClient = HttpClients.custom().setSSLSocketFactory(sslConFactory).build();
+            LOGGER.info("httpClient has been initialized");
+            ClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory(httpClient);
+            LOGGER.info("requestFactory has been initialized");
+            RestTemplate restTemplate = new RestTemplate(requestFactory);
+            LOGGER.info("restTemplate has been initialized");
+            return restTemplate;
+        } catch (Exception e) {
+            LOGGER.error("Failed to initialize RestTemplate with SSL context. Using a simple RestTemplate.", e);
+            return new RestTemplate();
```

Comment:
That seems suspicious to me. We try to use a template with a certificate, but for whatever reason we use a simple one? Isn't it the case that falling back to lower level should be done only in special cases/environments?